test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -179,3 +191,31 @@ def test_mask_along_axis_iid(self, mask_param, mask_value, axis):
 
         assert mask_specgrams.size() == specgrams.size()
         assert (num_masked_columns < mask_param).sum() == num_masked_columns.numel()
+
+
+@skipIfNoExec('sox')
+class ApplyCodecTestBase(TempDirMixin, TorchaudioTestCase):
+    backend = "sox_io"
+
+    def smoke_test(self, format, compression):
+        """
+        The purpose of this test suite is to verify that apply_codec functionalities do not exhibit
+        abnormal behaviors.
+        """
+        path = self.get_temp_path(f'data.{format}')
+        torch.random.manual_seed(42)
+        waveform = torch.rand(2, 44100 * 1)
+        sample_rate = 8000
+        augmented_data = F.apply_codec(waveform, sample_rate, format=format, channels_first=True,
+                                       compression=compression)
+        save_wav(path, augmented_data, sample_rate)
+        info = sox_io_backend.info(path)
+        assert info.sample_rate == sample_rate
+
+    @_mod_utils.requires_module('torchaudio._torchaudio')
```

Comment:
You can use `@skipIfNoExtension` instead of `@_mod_utils.requires_module`, and you can put it on the class definition so that you do not need to do it on every method.